frontend/src/abs/lang/abslang.abs

```diff
@@ -595,6 +598,26 @@ data HTTPCallableAnnotation = HTTPCallable;
 type Expansion = Int;
 type ExpansionCall = List<Int>;
 
+// Basic higher-order functions (map, filter, foldl and foldr)
+def List<B> map<A, B>(f)(List<A> list) = case list {
+    Nil => Nil;
+    Cons(x, xs) => Cons(f(x), map(xs));
+};
+
```

Comment:
Hmm, we already have a function `map` (the `Map` constructor function).  At the moment ABS doesn't have function overloading, so I'd expect a compile-time error here.

Incidentally, this last commit breaks the Java backend (the created file `Map.java` which defines the Map datatype is empty).  I think it's unlikely that this commit actually introduces the bug rather than just exposing it, but didn't investigate further yet which commit actually introduces the breakage.

Reply:
On second thought, since normal and partial functions are syntactically distinct, it might be ok to have them in separate namespaces.

Reply:
I also assume the namespaces won't collide. I'll look into the problem with the Java-Backend though

Reply:
The problem with the Java backend should be fixed in c7a24e8cdd30cf7367701172da4cc399c8ed3eec .